Mathlib/LinearAlgebra/TensorProduct/Subalgebra.lean

```diff
@@ -0,0 +1,203 @@
+/-
+Copyright (c) 2024 Jz Pan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jz Pan
+-/
+import Mathlib.LinearAlgebra.TensorProduct.Submodule
+import Mathlib.RingTheory.TensorProduct.Basic
+import Mathlib.LinearAlgebra.FreeModule.StrongRankCondition
+
+/-!
+
+# Some results on tensor product of subalgebras
+
+## Linear maps induced by multiplication for subalgebras
+
+Let `R` be a commutative ring, `S` be a commutative `R`-algebra.
+Let `A` and `B` be `R`-subalgebras in `S` (`Subalgebra R S`). We define some linear maps
+induced by the multiplication in `S`, which are
+mainly used in the definition of linearly disjointness.
+
+- `Subalgebra.mulMap`: the natural `R`-algebra homomorphism `A ⊗[R] B →ₐ[R] S`
+  induced by the multiplication in `S`, whose image is `A ⊔ B` (`Subalgebra.mulMap_range`).
+
+- `Subalgebra.mulMap'`: the natural `R`-algebra homomorphism `A ⊗[R] B →ₗ[R] A ⊔ B`
+  induced by multiplication in `S`, which is surjective (`Subalgebra.mulMap'_surjective`).
+
+-/
+
+open scoped Classical TensorProduct
+
+open FiniteDimensional
+
+noncomputable section
+
+universe u v w
+
+namespace Subalgebra
+
+variable {R : Type u} {S : Type v}
+
+section Semiring
+
+variable [CommSemiring R] [Semiring S] [Algebra R S]
+
+variable (A : Subalgebra R S)
```

Comment:
This generalizes to `A : Type` where `[IsScalarTower R A S]` or similar, right?

Reply:
Maybe yes; I'll check it later.

Reply:
I'm afraid that this is not true, for example `R=A=Z, S=F_p`. You may require that the map `A -> S` is injective. But in my opinion this is the same as `A : Subalgebra R S`. Another problem is that this isomorphism is defined in terms of `Submodule.lTensorOne`, which has no such generalizations (yet).